docs/topics/testing/advanced.txt

```diff
@@ -67,6 +67,35 @@ The following is a simple unit test using the request factory::
             response = MyView.as_view()(request)
             self.assertEqual(response.status_code, 200)
 
+.. _topics-testing-advanced-allowed-hosts:
+
+Tests and ``ALLOWED_HOSTS``
+===========================
+
+.. versionadded:: 1.11
+
+If your application includes unit tests with custom host names, those tests
```

Comment:
I'd explain the rationale and elaborate on why `ALLOWED_HOSTS = ['*']` isn't recommended. If I had such a project, this technique seems to introduce a lot of boilerplate, so I'd like to know the downsides to the settings file approach.